src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -375,33 +375,57 @@ using ContainsExpression =
 // STRAFTER / STRBEFORE
 template <bool isStrAfter>
 [[maybe_unused]] auto strAfterOrBeforeImpl =
-    [](std::string_view text, std::string_view pattern) {
-      // Required by the SPARQL standard.
-      if (pattern.empty()) {
-        return toLiteral(text);
-      }
-      auto pos = text.find(pattern);
-      if (pos >= text.size()) {
-        return toLiteral("");
-      }
-      if constexpr (isStrAfter) {
-        return toLiteral(text.substr(pos + pattern.size()));
-      } else {
-        // STRBEFORE
-        return toLiteral(text.substr(0, pos));
-      }
-    };
+    [](std::optional<ad_utility::triple_component::Literal> literal,
+       std::optional<ad_utility::triple_component::Literal> optPattern)
+    -> IdOrLiteralOrIri {
+  if (!optPattern.has_value() || !literal.has_value()) {
+    return Id::makeUndefined();
+  }
+  //  If the pattern has a language tag and the literal does not have the same
+  //  language tag, then throw an error.
+  const auto& pattern = asStringViewUnsafe(optPattern.value().getContent());
+  if ((optPattern.value().hasLanguageTag() &&
+       !literal.value().hasLanguageTag()) ||
+      (optPattern.value().hasLanguageTag() &&
+       literal.value().hasLanguageTag() &&
+       literal.value().getLanguageTag() !=
+           optPattern.value().getLanguageTag())) {
+    AD_THROW(
+        "The language tags of the literal and the pattern are incompatible.");
+  }
```

Comment:
This is wrong, the language tag of the pattern is not taken into account.
In the sparql standard, the pattern is a simple literal without any language tag. 
So the simplest thing is to just get the content pattern and that's that. or even simpler, just use the string value getter that directly gives the content for the pattern argument.